code/modules/reagents/chemistry/reagents/impure_reagents/impure_toxin_reagents.dm

```diff
@@ -0,0 +1,73 @@
+//Reagents produced by metabolising/reacting fermichems inoptimally these specifically are for toxins
+//Inverse = Splitting
+//Invert = Whole conversion
+//Failed = End reaction below purity_min
+
+////////////////////TOXINS///////////////////////////
+
+//Lipolicide
+//impure
+/datum/reagent/impurity/ipecacide
+	name = "Ipecacide"
+	description = "An extremely gross substance that induces vomiting. It is produced when Lipolicide reactions are impure."
+	ph = 7
+	liver_damage = 0
+
+/datum/reagent/impurity/ipecacide/on_mob_add(mob/living/carbon/owner)
+	if(owner.disgust >= DISGUST_LEVEL_GROSS)
+		return ..()
+	owner.adjust_disgust(50)
+	..()
+
+
+//Formaldehyde
+//impure
+/datum/reagent/impurity/methanol
+	name = "Methanol"
+	description = "A light, colourless liquid with a distinct smell. Ingestion can lead to blindness. It is a byproduct of organisms processing impure Formaldehyde."
+	reagent_state = LIQUID
+	color = "#aae7e4"
+	ph = 7
+
+/datum/reagent/impurity/methanol/on_mob_life(mob/living/carbon/owner)
+	var/obj/item/organ/eyes/eyes = owner.getorganslot(ORGAN_SLOT_EYES)
+	if(!eyes)
+		return ..()
+	eyes.applyOrganDamage(0.5)
```

Comment:
This should base itself on delta time.

Reply:
It is now based on delta time, tested and functional 👍